core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -1087,12 +1118,11 @@ object QualificationAppInfo extends Logging {
   }
 
   // Summarize and estimate based on wall clock times
-  def calculateEstimatedInfoSummary(estimatedRatio: Double, sqlDataFrameDuration: Long,
+  def calculateEstimatedInfoSummary(estimatedRatio: Double, sqlDataFrameDurationToUse: Long,
       appDuration: Long, sqlSpeedupFactor: Double, appName: String, appId: String,
       hasFailures: Boolean, mlSpeedupFactor: Option[MLFuncsSpeedupAndDuration] = None,
       unsupportedExecs: String = "", unsupportedExprs: String = "",
       allClusterTagsMap: Map[String, String] = Map.empty[String, String]): EstimatedAppInfo = {
-    val sqlDataFrameDurationToUse = wallClockSqlDataFrameToUse(sqlDataFrameDuration, appDuration)
 
     // get the average speedup and duration for ML funcs supported on GPU
     val (mlSpeedup, mlDuration) = if (mlSpeedupFactor.isDefined) {
```

Comment:
this comment is for  the below code.

speedUpOpportunitySQL should just be the supportedSqlDfWallDuration. We've already converted to wall clock, the estimatedRatio in the original code was intended to do that conversion.  
I don't know about the ml code here so we need to check to see if that is already wall clock here?  From a quick glance it doesn't look like it is.

The other ones we are trying to fill in:
   ```
 estimatedGpuDur: Double, // Predicted runtime for the app if it was run on the GPU
    estimatedGpuSpeedup: Double, // app_duration / estimated_gpu_duration
    estimatedGpuTimeSaved: Double, // app_duration - estimated_gpu_duration
```
estimatedGpuDur - I think this stays the same at (speedUpOpportunitySQL / sqlSpeedupFactor) +
      estimated_wall_clock_dur_not_on_gpu + (mlDuration / mlSpeedup)..    but kind of seems like we are adding mlDuration in twice. We need to make sure estimated_wall_clock_dur_not_on_gpu here doesn't include the mlDuration.  

I think estimatedGpuSpeedup and estimatedGpuTimeSaved stay as the formula says

Reply:
> I don't know about the ml code here so we need to check to see if that is already wall clock here?
It is wallclock duration for the aggregated over all the stages having ML functions.